### DISCUSSION TITLE
x_r and x_l. What does that stand for?

### DISCUSSION THREAD

**Discussion by Jeriousman** [pyg-team/pytorch_geometric#4008] [Q&A] [answered]
I know x_j and x_i convention. But When I encounter x_r and x_l in the source code. I do not know what it really means and there is no explicit explanation. Can you describe what it means?

**Comment by rusty1s** [comment, marked as answer]
Yeah, sorry for that. I'm not so proud of this naming as well :) In general, it just means that we separate between transforming source (`x_l`) (for left) and destination nodes (`x_r`) (for right). For example, in bipartite graphs, `x = (x_l, x_r)` may be given as a tuple. A better naming convention would be `x_src = x_l` and `x_dst = x_r`.

**Reply by Jeriousman** [reply]
Thank you for the clarification!
